File: 2/js/src/MoveElementToEnd.ts
function swap(i: number, j: number, list: number[]) {
	const temp = list[i];
	list[i] = list[j];
	list[j] = temp;
}

function solve(list: number[], toMove: number) {
	const n = list.length;

	let i=0;
	let j=n-1;

	while(i < n && i < j) {
		if(list[i] == toMove) {
			while(list[j] == toMove && i < j) {
				j--;
			}

			swap(i, j, list);
			j--;
		}

		i++;
	}
}

(function main() {
	const list = [2,1,2,2,2,3,4,2];
	const toMove = 2;

	solve(list, toMove);
	console.log(list);
})();

File: 2/java/MoveElementToEnd.java
class MoveElementToEnd {
	static void moveElemToEnd(int arr[], int toMove) {

		int i = 0;
		int j = arr.length -1;

		while(i < j) {
			if(arr[i] == toMove) {
				while(arr[j] == toMove) {
					j--;	

					if(j <= i) {
						break;
					}
				}

				int temp = arr[i];
				arr[i] = arr[j];
				arr[j] = temp;
				j--;
			}
			i++;
		}
	}

	public static void main(String args[]) {
		int arr[] = {2,1,2,2,2,3,4,2};
		int toMove = 2;

		moveElemToEnd(arr, 2);

		for(int i : arr) {
			System.out.print(i + " ");
		}
		System.out.println();
	}
}
